sa/sa.go

```diff
@@ -2216,3 +2216,70 @@ func (ssa *SQLStorageAuthority) SerialsForIncident(req *sapb.SerialsForIncidentR
 	}
 	return nil
 }
+
+// GetRevokedCerts gets a request specifying an issuer and a period of time,
+// and writes to the output stream the set of all certificates issued by that
+// issuer which expire during that period of time and which have been revoked.
+func (ssa *SQLStorageAuthority) GetRevokedCerts(req *sapb.GetRevokedCertsRequest, stream sapb.StorageAuthority_GetRevokedCertsServer) error {
+	atTime := time.Unix(0, req.RevokedBefore)
+
+	// TODO: Analyze and optimize this query.
+	// Should we condition on `WHERE status = 'revoked'`?
+	// Should we condition on `WHERE revokedDate < atTime`?
+	// I think the answer to both of the above is no, since we don't have indexes
+	// on either of those columns.
+	query := `SELECT serial, status, revokedReason, revokedDate
```

Comment:
The approach you use there -- confirming that the rows returned by a query have columns which match what we expect -- makes total sense to me when doing a `SELECT * FROM`. But here we're explicitly listing the columns we want, which means (imo) that we can trust that exactly those columns will be returned -- if they don't exist, we'll get an error.

I think the killer feature here would be if using gorp in this way allowed us to read the row directly into a struct, rather than having to explicitly list each data member of that struct in the call to `rows.Scan(...)`, just like we can with the non-streaming [`dbMap.Select`](https://pkg.go.dev/gopkg.in/gorp.v2#DbMap.Select). That would make it impossible to break the query by accidentally editing the `SELECT` or the `Scan` but not both. But I'm not aware of any way to do that.

So I'm not sure that all of the extra column confirmation gets us much, here? If the column types don't match the fields they're read into in the `Scan` call, that'll be an error. But confirming that the column names are what we expect doesn't prevent the `Scan` call from accidentally being out of order, so it doesn't seem to provide much protection to me.

Reply:
Yeah, I can definitely see where you're coming from here. This solution misses out on a lot of guarantees that you could gain from pursuing the pattern in `ScanCertStatusMetadataRow()`.

(we decided, offline, to implement one of these patterns and eventually fork Gorp to introduce a query-and-scan method that will give us the same guarantees that we get from `dbmap.Select()`.